kpi/filters.py

```diff
@@ -177,22 +152,7 @@ def filter_queryset(self, request, queryset, view):
         she should see all permissions for that object, including those
         assigned to other users.
         """
-        possible_content_types = ContentType.objects.get_for_models(
-            *get_models_with_object_permissions()
-        ).values()
-        result = queryset.none()
-        for content_type in possible_content_types:
-            # Find all the permissions assigned to the user
-            permissions_assigned_to_user = ObjectPermission.objects.filter(
-                content_type=content_type,
-                user=user,
-            )
-            # Find all the objects associated with those permissions, and then
-            # find all the permissions applied to all of those objects
-            result |= ObjectPermission.objects.filter(
-                content_type=content_type,
-                object_id__in=permissions_assigned_to_user.values(
-                    'object_id'
-                ).distinct()
-            )
+        result = ObjectPermission.objects.filter(
+            asset__permissions__user=user
```

Comment:
Something I'm wondering: Is Django is smart enough to not add a join on `auth_user`  because of this condition `asset_permissions__user=user` ? 
Should it be better to use user's PK directly. e.g. `asset_permissions_user_id=user.pk`

Reply:
@noliveleger, yes, Django is smart enough. Let's not clutter our code with `…_id=….pk` where it's not needed:
```
>>> ObjectPermission.objects.filter(asset__permissions__user=user).distinct().count()
SELECT COUNT(*)
  FROM (
        SELECT DISTINCT "kpi_objectpermission"."id" AS Col1,
               "kpi_objectpermission"."user_id" AS Col2,
               "kpi_objectpermission"."permission_id" AS Col3,
               "kpi_objectpermission"."deny" AS Col4,
               "kpi_objectpermission"."inherited" AS Col5,
               "kpi_objectpermission"."asset_id" AS Col6,
               "kpi_objectpermission"."uid" AS Col7
          FROM "kpi_objectpermission"
         INNER JOIN "kpi_asset"
            ON ("kpi_objectpermission"."asset_id" = "kpi_asset"."id")
         INNER JOIN "kpi_objectpermission" T3
            ON ("kpi_asset"."id" = T3."asset_id")
         WHERE T3."user_id" = 1
       ) subquery


Execution time: 0.006980s [Database: default]

888
```

There are definitely still places where `object.related_id` is the way to go in order to avoid an extra query:
```
# a = Asset.objects.last()
>>> a.owner_id
21
>>> a.owner.pk
SELECT "auth_user"."id",
       "auth_user"."password",
       "auth_user"."last_login",
       "auth_user"."is_superuser",
       "auth_user"."username",
       "auth_user"."first_name",
       "auth_user"."last_name",
       "auth_user"."email",
       "auth_user"."is_staff",
       "auth_user"."is_active",
       "auth_user"."date_joined"
  FROM "auth_user"
 WHERE "auth_user"."id" = 21


Execution time: 0.000461s [Database: default]

21
```